packages/next/src/elements/LeaveWithoutSaving/usePreventLeave.tsx

```diff
@@ -1,6 +1,7 @@
 // Credit: @Taiki92777
 //    - Source: https://github.com/vercel/next.js/discussions/32231#discussioncomment-7284386
 // Credit: `react-use` maintainers
+import { useModal } from '@payloadcms/ui'
```

Comment:
This hook should not rely on useModal whatsoever, if you need to performa. side-effect, pass in your callback function as a prop.